apisix/plugins/ai-proxy/drivers/openai.lua

```diff
@@ -0,0 +1,83 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local _M = {}
+
+local core = require("apisix.core")
+local http = require("resty.http")
+
+local pairs = pairs
+
+-- globals
+local DEFAULT_HOST = "api.openai.com"
+local DEFAULT_PORT = 443
+
+
+function _M.request(conf, request_table, ctx)
+    local httpc, err = http.new()
+    if not httpc then
+        return nil, "failed to create http client to send request to LLM server: " .. err
+    end
+    httpc:set_timeout(conf.timeout)
+
+    local custom_host = core.table.try_read_attr(conf, "override", "host")
+    local custom_port = core.table.try_read_attr(conf, "override", "port")
+    local custom_path = core.table.try_read_attr(conf, "override", "path")
+    local custom_scheme = core.table.try_read_attr(conf, "override", "scheme")
```

Comment:
Would it be better to just use a combined ENDPOINT configuration? 
I.e. `override.endpoint = "https://api.openai.com:443/api-balabala/"`

In this way schema, host, and port can be combined into a single configuration. If we do need such a separate section later, APISIX can parse it internally instead of asking the user to configure it one by one.

Reply:
you are right